lisa/tests/kernel/scheduler/load_tracking.py

```diff
@@ -733,3 +735,218 @@ def test_load_avg_behaviour(self, error_margin_pct=5, allowed_error_pct=5)\
+        Test that a migrated task properly propagates its utilization at the CPU level
+
+        :param allowed_error_pct: How much the trace averages can stray from the
+          expected values
+        :type allowed_error_pct: float
+        """
+        expected_cpu_util = self.get_expected_cpu_util()
+        trace_cpu_util = self.get_trace_cpu_util()
+
+        passed = True
+
+        expected_metrics = {}
+        trace_metrics = {}
+        deltas = {}
+
+        for cpu in self.cpus:
+            cpu_str = "cpu{}".format(cpu)
+
+            expected_metrics[cpu_str] = TestMetric({})
+            trace_metrics[cpu_str] = TestMetric({})
+            deltas[cpu_str] = TestMetric({})
+
+            for phase in range(self.nr_phases):
+                if not self.is_almost_equal(
+                        trace_cpu_util[cpu][phase],
+                        expected_cpu_util[cpu][phase],
+                        allowed_error_pct):
+                    passed = False
+
+                # Just some verbose metric collection...
+                phase_str = "phase{}".format(phase)
+
+                expected_metrics[cpu_str].data[phase_str] = TestMetric(
+                    expected_cpu_util[cpu][phase])
+                trace_metrics[cpu_str].data[phase_str] = TestMetric(
+                    trace_cpu_util[cpu][phase])
+                deltas[cpu_str].data[phase_str] = TestMetric(
+                    -100 * (expected_cpu_util[cpu][phase] - trace_cpu_util[cpu][phase]) /
+                    expected_cpu_util[cpu][phase], "%")
+
+        res = ResultBundle.from_bool(passed)
+        res.add_metric("Expected utilization", expected_metrics)
+        res.add_metric("Trace utilization", trace_metrics)
+        res.add_metric("Utilization deltas", deltas)
+
+        return res
+
+class OneTaskCPUMigration(CPUMigrationBase):
+    """
+    Some tasks on two big CPUs, one of them migrates in its second phase.
+    """
+
+    @classmethod
+    def get_migration_cpus(cls, te):
+        cpu_capacities = te.plat_info["cpu-capacities"]
+
+        return [cpu for cpu, capa in cpu_capacities.items()
+                if capa == max(cpu_capacities.values())][:2]
+
+    @classmethod
+    def check_from_testenv(cls, te):
+        super().check_from_testenv(te)
+
+        cpus = cls.get_migration_cpus(te)
+        if not len(cpus) == 2:
+            raise RuntimeError("This workload requires 2 CPUs of highest capacity")
+
+    @classmethod
+    def get_rtapp_profile(cls, te):
+        profile = {}
+        cpus = cls.get_migration_cpus(te)
+
+        for task in ["migrating", "static0", "static1"]:
+            # An empty RTATask just to sum phases up
+            profile[task] = RTATask()
+
+        for i in range(2):
+            # A task that will migrate to another CPU
+            profile["migrating"] += Periodic(
+                duty_cycle_pct=20, duration_s=cls.PHASE_DURATION_S,
+                period_ms=cls.TASK_PERIOD_MS, cpus=[cpus[i]])
```

Comment:
can we pass in the calibration cpu here ?
In that case `get_migration_cpus()` could be update to return the highest capacity pair of CPUs in the system...

Reply:
As discussed offline, I'd rather change the duty cycles than affect the calibration, because otherwise 20% duty_cycle != 20% utilization.